lib/internal/process/pre_execution.js

```diff
@@ -650,6 +655,22 @@ function markBootstrapComplete() {
   internalBinding('performance').markBootstrapComplete();
 }
 
+// Sequence number for diagnostic filenames
+let sequenceNumOfheapSnapshot = 0;
+
+function getHeapSnapshotFilename(diagnosticDir) {
+  if (!diagnosticDir) return undefined;
+
+  const date = new Date();
+  const formattedDateString = DatePrototypeToLocaleDateString(date).split('/').reverse().join('');
```

Comment:
This gives me something like `2023195` for my locale, not `20230519` as it's supposed to be. Fixing the locale would require intl support. To get it properly from JS land we'd need to use the `getFullYear()` methods etc.. Though we could also simply reuse `DiagnosticFilename` from the C++ land if we simply tweak the `triggerHeapSnapshot` binding a bit to accept an optional directory.

Reply:
I don't think we need to edit the cpp functions here, instead, we can specify the locale so we don't need the dependency of getFullYear, month, day, etc... 


So, it will work regardless of the locale. I will push the changes soon the next time I access my Mac.

Reply:
However, since we have many related functions in primordials,  we can make use of it.